### DISCUSSION TITLE
FiveM

### DISCUSSION THREAD

**Discussion by ShyRoxxx** [MarkusJx/autobet#58] [Q&A]
Hey, is it possible to use in FiveM ?

**Comment by MarkusJx** [comment]
Yes, maybe. I don't know. Just try it out and you'll see :)

As this is just a glorified screen reader combined with a auto clicker, it should work with any application having the same UI as the native GTA Online Horse racing implementation.

You may need to change the application name, as it may not be 'GTAV.exe', but that can be done from inside the UI.

**Reply by ShyRoxxx** [reply]
Yeah, i tried but i cant select fivem as a application :( Is there any way to change this ?

**Reply by MarkusJx** [reply]
Can you open the application selector dropdown? If you can't you've already started the betting process and just have to stop it.

Or is the Application not in the list of Apps to select? The Application may have a different name from what you'd expect. Try selecting the application in the Task Manager and find out the name of the executable. Or the dimensions of the FiveM	window are not valid for some reason, in this case, it won't work.

And make sure FiveM is actually running and In-Game, otherwise none of this will work. You may need to change into windowed mode and not minimize it for the application to  show up

**Reply by ShyRoxxx** [reply]
So I can open the dropdown menu and also select applications. When I then click on the Application Fivem incl. the server I am on it tells me that my game is not running even though I am ingame on the side.

**Reply by ShyRoxxx** [reply]
<img width="2557" alt="image" src="https://user-images.githubusercontent.com/114837756/208424689-1f5ce558-c24f-4489-a624-196d4c3752f9.png">

**Reply by ShyRoxxx** [reply]
Looks so sceetchy because the screenshot goes over 2 screens

**Reply by MarkusJx** [reply]
Makes sense. Can you create a log from application startup until this point by following [these instructions](https://github.com/MarkusJx/autobet/wiki/Debugging#debugging-using-the-debug-option)?

**Reply by ShyRoxxx** [reply]
Sure

**Reply by ShyRoxxx** [reply]
Cant find the Directory. I am currently checking my C:\Users\Shyroxx\AppData\Local\Programs\autobet folder but there is not Log file

**Reply by MarkusJx** [reply]
Try stopping the application or disable logging, it should be created after that

**Reply by ShyRoxxx** [reply]
Nope. Nothing happens

**Reply by ShyRoxxx** [reply]
Console LOG 
<img width="1204" alt="image" src="https://user-images.githubusercontent.com/114837756/208427849-73185e4f-2e82-4c91-89f0-422f6fef7e75.png">

**Reply by MarkusJx** [reply]
This is interesting. It can't read the settings, which may cause them to be reset. Also, the wiki page is outdated, the logs should be located at `C:\Users\YOUR_USER_NAME\Documents\autobet`. This is also where the settings should be located at. Do you have rights to access your documents folder? It seems like an issue with access rights

**Reply by ShyRoxxx** [reply]
Yea i got rights to it. I send you the Logs 
[autobet.log](https://github.com/MarkusJx/autobet/files/10259313/autobet.log)

**Reply by MarkusJx** [reply]
According to the logs, it can read the settings but only if the application name is not changed. So I'm assuming, there are non-UTF-8 characters in the window name (as shown by the �).

Unless there is an option inside FiveM (or windows) to change the window title, I'm not able to help, sorry. This is because I'd need to change a lot of stuff to utf-16, which may break other things and still won't guarantee this will work. I may run into other issues with the win32 API not being able to handle those other strings, and although fixable, that's not a thing I'm willing to do, considering this game is almost 10 Years old and will not receive any further updates.

**Reply by ShyRoxxx** [reply]
So its not really possible for FiveM ?

**Reply by MarkusJx** [reply]
I mean you could try changing the window title using [something like this](https://superuser.com/a/1579524), but that would also be annoying.

Also, what is the character in place of the � symbol, are there emojis in the window's title or what the hell is going on in there?

**Reply by ShyRoxxx** [reply]
Yeah there are 2 Emojis in the title. ill try the link u send

**Reply by ShyRoxxx** [reply]
Oh man hab jetzt erst gesehen das du auch deutsch bist xD

**Reply by ShyRoxxx** [reply]
Verstehe leider nicht viel vom Coden und weiß nicht wie ich [das](https://superuser.com/questions/1579485/how-to-change-the-title-window-of-a-window-in-windows-10/1579524#1579524) richtig benutzen sollte

**Reply by MarkusJx** [reply]
Yeah, that's crazy, right, Germany has more than one citizen. But let's keep this conversation English-only, so any non-Germans will be able to understand everything..

And why are there Emojis in the Application's title? Tell the server owners to grow up, that'd make my life much easier :D
(Says the guy who works for the company which invented the emoji, but what do I know)

**Reply by ShyRoxxx** [reply]
So is it that there is no other workaround except [that](https://superuser.com/questions/1579485/how-to-change-the-title-window-of-a-window-in-windows-10/1579524#1579524) ?

**Reply by ShyRoxxx** [reply]
But when i am able to change the window name it should be possible ?

**Reply by MarkusJx** [reply]
Yes, well, maybe. [This](https://stackoverflow.com/a/39033389) has better instructions than the other one, maybe try this

**Reply by ShyRoxxx** [reply]
> Yes, well, maybe. [This](https://stackoverflow.com/a/39033389) has better instructions than the other one, maybe try this

This is much better. Thanks 💯

**Reply by ShyRoxxx** [reply]
But it wont change ? I testet the script on other Applications where it worked, but on FiveM it dont seems to work or did i just typed the wrong name in ?? The tut says that i need to put the applicationnames etc in it. So i typed in "%userprofile%\desktop\settext" "FiveM_b2699_GTAProcess.exe - FiveM  by Cfx.re - CountySide.net ?? GER|Roleplay|Whitelist|discord.gg/countyside" "Test" but it wont work ? Any clue ?

**Reply by MarkusJx** [reply]
Huh, maybe that doesn't work after all. Either it's a completely different issue or my program is going after the actual window title, which can't be changed.

**Reply by ShyRoxxx** [reply]
Could it also be due to characters like ® that it does not work properly?

**Reply by MarkusJx** [reply]
No, don't think so. If you select the program in the ui with the new title, does it display the new title or still the old one?

**Reply by ShyRoxxx** [reply]
Nah, the Window Title doesn't changes.

**Reply by MarkusJx** [reply]
Yeah, that's what I thought. I may look into making this thing "emoji-compatible". Don't know yet. Will post an update once I know. May happen tomorrow.

**Reply by ShyRoxxx** [reply]
That would be very nice

**Reply by MarkusJx** [reply]
I've created a new issue in order to track this more efficiently: https://github.com/MarkusJx/autobet/issues/59

**Reply by MarkusJx** [reply]
As the build finally succeeded, you may want to try out the latest beta build by downloading it [here](https://github.com/MarkusJx/autobet/suites/9995006319/artifacts/485211599) and installing the binary.

**Reply by ShyRoxxx** [reply]
Yeah, it recognizes the Application. I didn't won any round yet, but the Autoclicker is working :)

**Reply by ShyRoxxx** [reply]
But i think, the screenreader got an error. 
<img width="699" alt="image" src="https://user-images.githubusercontent.com/114837756/209001033-250adfbf-b828-4d38-a51a-0e68ddc4a6c9.png">

I just bid in 1k steps

**Reply by MarkusJx** [reply]
Please consult the [betting rules](https://github.com/MarkusJx/autobet/wiki/Betting-rules) first.
If you think the Program isn't working as expected (e.g. the amounts placed are too low) try increasing the "Time sleep" or "Click sleep" values. If there are still issues, send me some logs and I can take a look at this

**Reply by ShyRoxxx** [reply]
So if i start with 100k and let it do his work for 1 h it schould be around 200 K ?

**Reply by ShyRoxxx** [reply]
with 10k steps

**Reply by ShyRoxxx** [reply]
I think the HTML is broken
<img width="610" alt="image" src="https://user-images.githubusercontent.com/114837756/209019488-cea0359d-5c2f-4591-ad00-cb83d8c70757.png">

**Reply by ShyRoxxx** [reply]
Could you recommend any settings to make Profit ?

**Reply by MarkusJx** [reply]
~~There are no settings to change~~. It may just work™.

**Reply by MarkusJx** [reply]
If the bets are placed correctly, then you should make a decent profit unless FiveM or your server changed the implementation of the betting stuff. Oh, and you can totally change the betting process, via betting functions. There's a wiki page on that. But you'll need to figure out the best setting for your situation. Or you could generate statistics and du the math yourself. I could also use some kind if machine learning to figure out the best settings but I definitely won't do that.

**Reply by ShyRoxxx** [reply]
Yeah,I got some statistics but don't now how to use it haha. How could I edit it, if I want to bet on the higher horses like 13/1 etc. Because I won a lot of games with Kraff Running (14/1)

**Reply by MarkusJx** [reply]
Just take a look at the [wiki page for the betting functions](https://github.com/MarkusJx/autobet/wiki/Custom-betting-functions). So basically, what you'd have to do is change the default implementation to return a high-risk value:
```js
// Convert an odd string such as '2/1'
// to a number, e.g. 2
function toNumber(val) {
    return Number(val.split('/')[0]);
}

// Main function
// Returns string, the odd of the horse to bet on,
// or null, if no bet should be placed
function run() {
	// This is the interesting part:
	// Bet on "14/1" if it exists
	if ("14/1" in odds) {
		return "14/1";
	}

    let containsEvens = odds.includes("evens");
    let lowest = null;

    for (let i = 0; i < odds.length; i++) {
        // Check if one probability >= 5/1 exists multiple times
        for (let j = i + 1; j < odds.length; j++) {
            // If odds[i] >= 5/1 exists multiple times, do not bet
            if ((odds[i] == "evens" || toNumber(odds[i]) <= 5) && odds[i] == odds[j])
                return null;
        }

        // If odds contains 'evens' and does also
        // contain another one >= 3/1, do not bet
        if (containsEvens && (odds[i] == "2/1" || odds[i] == "3/1"))
            return null;

        // Set the lowest odd
        if (!containsEvens && (lowest == null || toNumber(odds[i]) < toNumber(lowest))) {
            lowest = odds[i];
        }
    }

    // If the odds contain evens, this is the lowest
    if (containsEvens) {
        lowest = "evens";
    }

    // Return the lowest odd
    return lowest;
}

// Set the result.
// Set null if no bet should be placed,
// the odd of the horse to bet on otherwise
setResult(run());
```

But statistically speaking, it's not a good move to bet on those high-risk horses as you most likely will lose. The default implementation will most likely guarantee you getting steadily at least some money (atound 200k per hour).

**Reply by ShyRoxxx** [reply]
Yea, ill try it again with the default settings. Starting with 100k and 10 k steps. Hope i make some Profit

**Reply by ShyRoxxx** [reply]
I think the beta Version is a bit buggy. Just lost about 170k with default profile

**Reply by MarkusJx** [reply]
Did you check the behavior of the program against the betting rules? As I didn't change anything about the betting, I won't do anything as long as you won't prove illicit behavior.

**Reply by MarkusJx** [reply]
As I felt like this may be an important issue that needs to be addressed, I've added a [gambling PSA](https://github.com/MarkusJx/autobet#important-notes) in the docs.

**Reply by MarkusJx** [reply]
But if you feel like you lost an unreasonable amount of money, here are some steps for you to follow (in order):
1. Consult the [betting rules](https://github.com/MarkusJx/autobet/wiki/Betting-rules). If there seems to be an issue, please let me know. But if the behavior of the program complies with these rules, follow the next steps.
2. Consult the server owners if they changed anything about the betting system
3. Consult the authors of FiveM if they changed anything about the betting system
4. Try running the program again, it is designed to make money in the long run. As this is gambling, you might lose money in a short period of time, but gain some after longer durations.

The last point is not actual advice, though. If you still lose money after ~3 hours, stop using this software. There might be issues with the implementations of the betting stuff in FiveM or the server you're on.

Again, this is a gambling software. You shouldn't use it if you have prior incidents of gambling addiction or if you are underage. In both of those cases, stay away from the in-Game casino or don't play this game.

There is nothing wrong with this program as long as you gain money, but once you lose, we enter the territory of *actual gambling*, which is terrifying and extremely dangerous. Just keep that in mind.

That being said, you are solely responsible for everything you do with this software.

**Reply by ShyRoxxx** [reply]
Hey, last question I got. What are "Evens" so how are they defined ?

**Reply by MarkusJx** [reply]
There is an info page in-Game about the different odds, but I wouldn't know what it says, because I've never read that text, didn't seem important. And it probably isn't.

So you should probably read that
